### PR TITLE
refactor configuration (settings)

### PR DESCRIPTION
**Pull request by Haokun-Chen**
##### SUMMARY

- Refactor configuration page to be settings page
- create reusable card view directive
- add a settings-card view page and route to wrap on top of previous configuration sub-forms
- add temporary settings side-nav popover dropdown navigation as sub-menu 

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

 - UI

##### ADDITIONAL INFORMATION

previous configuration page structure:
1. configuration route -- configuration controller -- configuraiton.partial
2. sub-form serve under the configuration state: as child-view@configuration 

new settings page structure:
1. settings route -- settings controller -- settings.partial (card view), each card navigates to settings.form state, and passes the corresponding stateParams.
2. settings-form route -- settings-form controller -- settings-form.partial (wrapper handle all sub-form, similar to previous configuration page)
3. sub-form serve under settings-form state: as child-view@settings.form
4. combine license route into settings.form.route

### CODE REVIEW

File: awx/ui/client/lib/components/cards/_index.less [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+.twr-panel-text {
+    font-size: 0.9em;
+    color: #161b1f;
+}
+
+
+/* Card */
+
+.twr-card-container {
+    display: grid;
```
Comment (jaredevantabor) [inline awx/ui/client/lib/components/cards/_index.less]: This is cool 😎

File: awx/ui/client/lib/components/cards/_index.less [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+.twr-panel-title {
+    font-size: 1em;
+    color: #707070;
+    font-weight: bold;
+    line-height: 1em;
+}
+
+.twr-panel-text {
+    font-size: 0.9em;
+    color: #161b1f;
```
Comment (jaredevantabor) [inline awx/ui/client/lib/components/cards/_index.less]: we typically don't hard-code hex values, but rather we refer to them with a .less alias. The aliases are kept in `awx/ui/client/src/shared/branding/colors.default.less`

File: awx/ui/client/lib/components/cards/_index.less [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+/* Card */
+
+.twr-card-container {
+    display: grid;
+    grid-template-columns: repeat(auto-fit, minmax(200px, 1fr));
+    grid-gap: 20px;
+}
+
+.twr-card {
+    display: flex;
```
Comment (jaredevantabor) [inline awx/ui/client/lib/components/cards/_index.less]: My understanding of CSS-Grid is minimal at best, but if the parent container, in this case `twr-card-container` uses `display:grid`, then does it's child elements need to specify `display:flex`? I thought that the parent element would take care of all that for you.

Comment (jaredevantabor) [reply awx/ui/client/lib/components/cards/_index.less]: Seems kind of redundant to have css-grid and flex box in motion here. Maybe the flex is for the content in the card itself?

File: awx/ui/client/lib/components/cards/_index.less [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    padding: 20px;
+    border-left: 3px solid #ffffff;
+    border-radius: 2px;
+    box-shadow: 0px 3px 1px -2px rgba(0, 0, 0, 0.2), 0px 2px 2px 0px rgba(0, 0, 0, 0.14), 0px 1px 5px 0px rgba(0, 0, 0, 0.12);
+    transition: box-shadow 280ms cubic-bezier(0.4, 0, 0.2, 1);
+    cursor: pointer;
+}
+
+.twr-card:hover {
+    background-color: #fcfcfc;
```
Comment (jaredevantabor) [inline awx/ui/client/lib/components/cards/_index.less]: You could probably replace this w/ an alias as well.

File: awx/ui/client/lib/components/cards/_index.less [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    border-left: 3px solid #337ab7;
```
Comment (jaredevantabor) [inline awx/ui/client/lib/components/cards/_index.less]: and this hex too :)

File: awx/ui/client/lib/components/cards/_index.less [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    box-shadow: 0px 2px 4px -1px rgba(0, 0, 0, 0.2), 0px 4px 5px 0px rgba(0, 0, 0, 0.14), 0px 1px 10px 0px rgba(0, 0, 0, 0.12);
```
Comment (jaredevantabor) [inline awx/ui/client/lib/components/cards/_index.less]: is this the best way to write this?

File: awx/ui/client/lib/components/cards/_index.less [hunk 7]
```diff
@@ -0,0 +1,52 @@
+.content-padding {
+    padding: 20px;
+}
+
+/* Panels */
+.twr-panel-title {
+    font-size: 1em;
```
Comment (jaredevantabor) [inline awx/ui/client/lib/components/cards/_index.less]: I haven't seen us use `em` for font size much, typically `px`

File: awx/ui/client/src/configuration/auth-form/configuration-auth.controller.js [hunk 8]
```diff
@@ -63,27 +63,36 @@ export default [
...
         var activeAuthForm = 'azure';
+        var activeTab = 'azure';
         var ldapDropdownValue = '';
+        var githubDropdownValue = 'github';
 
         let codeInputInitialized = false;
 
         // Default active form
-        if ($stateParams.currentTab === '' || $stateParams.currentTab === 'auth') {
+        // if ($stateParams.currentTab === '' || $stateParams.currentTab === 'auth') {
```
Comment (jaredevantabor) [inline awx/ui/client/src/configuration/auth-form/configuration-auth.controller.js]: can remove commented out code

File: awx/ui/client/src/configuration/auth-form/configuration-auth.partial.html [hunk 9]
```diff
@@ -1,32 +1,47 @@
...
+                ng-model="authVm.githubDropdownValue"
+                ng-options="opt.value as opt.label for opt in authVm.githubDropdownOptions"
+                ng-change="authVm.activeForm('github')">
             </select>
         </div>
     </div>
 
-    <hr ng-show="authVm.dropdownValue === 'ldap'"/>
+
+    <!-- <hr ng-show="authVm.activeTab === 'ldap' || authVm.activeTab === 'github'"/> -->
```
Comment (jaredevantabor) [inline awx/ui/client/src/configuration/auth-form/configuration-auth.partial.html]: could remove

File: awx/ui/client/src/configuration/jobs-form/configuration-jobs.controller.js [hunk 10]
```diff
@@ -139,9 +143,9 @@ export default [
 
         // Fix for bug where adding selected opts causes form to be $dirty and triggering modal
         // TODO Find better solution for this bug
-        $timeout(function(){
-            $scope.$parent.configuration_jobs_template_form.$setPristine();
-        }, 1000);
+        // $timeout(function(){
+        //     $scope.$parent.configuration_jobs_template_form.$setPristine();
+        // }, 1000);
```
Comment (jaredevantabor) [inline awx/ui/client/src/configuration/jobs-form/configuration-jobs.controller.js]: if you don't think this hack is still necessary let's just remove it

File: awx/ui/client/src/configuration/jobs-form/configuration-jobs.controller.js [hunk 11]
```diff
@@ -174,8 +178,8 @@ export default [
             tab = parts[parts.length - 1];
 
             if (tab === 'jobs' && !codeInputInitialized) {
-                initializeCodeInput();
-                codeInputInitialized = true;
+                // initializeCodeInput();
+                // codeInputInitialized = true;
```
Comment (jaredevantabor) [inline awx/ui/client/src/configuration/jobs-form/configuration-jobs.controller.js]: can remove

File: awx/ui/client/src/configuration/license.route.js [hunk 12]
```diff
@@ -9,7 +9,7 @@ import { N_ } from '../i18n';
 import _ from 'lodash';
 
 export default {
-	name: 'configuration.license',
+	name: 'form.license',
 	route: '/license',
 	// templateUrl: templateUrl('license/license'),
```
Comment (jaredevantabor) [inline awx/ui/client/src/configuration/license.route.js]: can remove

File: awx/ui/client/src/configuration/license.route.js [hunk 13]
```diff
@@ -9,7 +9,7 @@ import { N_ } from '../i18n';
 import _ from 'lodash';
 
 export default {
-	name: 'configuration.license',
+	name: 'form.license',
```
Comment (jaredevantabor) [inline awx/ui/client/src/configuration/license.route.js]: shouldn't this be `settings.license`?

File: awx/ui/client/src/configuration/license.route.js (hunk 12 above)
Comment (jaredevantabor) [inline awx/ui/client/src/configuration/license.route.js]: can you remove these lines?

**Review by jaredevantabor — CHANGES_REQUESTED** [review]
So far, I've found some cleanup that could happen with the .less file, some comments that could be removed, and also, I can't get to the license screen. Something weird going on with that route. 

But I did pull it down again and it looks like saving/editing settings are still in tact which is good.

**Comment by wenottingham** [comment]
Apologies, this now has conflicts. Can you rebase and fix the conflicts?

**Comment by Haokun-Chen** [comment]
Sure! resolving conflicts

**Review by jaredevantabor — APPROVED** [review]

**Pull request merged**